src/app/services/products.service.ts

```diff
@@ -0,0 +1,181 @@
+import { ToastyNotificationsService } from './toasty-notifications.service';
+import { Injectable, EventEmitter } from '@angular/core';
+import { Router } from '@angular/router';
+import { HttpClient } from '@angular/common/http';
+import { environment } from '../../environments/environment';
+import { Observable } from 'rxjs';
+import 'rxjs/add/operator/map';
+
+
+import { Product } from '../products/product.model';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProductsService {
```

Comment:
most of this class will match what wesley has on his branch

Reply:
Yes